src/device/socket/vsock.rs

```diff
@@ -86,13 +97,40 @@ impl<'a, H: Hal, T: Transport> VirtIOSocket<'a, H, T> {
             tx,
             event,
             guest_cid,
-            queue_buf_dma,
-            queue_buf_rx,
+            rx_buf_dma,
+            rx_buf,
+            connection_info: None,
         })
     }
 
+    fn fill_rx_queue(
+        rx: &mut VirtQueue<H, { QUEUE_SIZE }>,
+        rx_buf: &mut [u8],
+        transport: &mut T,
+    ) -> Result {
+        let buffer_size = rx_buf.len() / QUEUE_SIZE;
+        if buffer_size < size_of::<VirtioVsockHdr>() {
+            return Err(SocketError::BufferTooShort.into());
+        }
+        for i in 0..QUEUE_SIZE {
+            let start = i * buffer_size;
+            let end = start + buffer_size;
+            // Safe because the buffer lives as long as the queue.
```

Comment:
I don't think you know that here. The reference being passed in only has a lifetime as long as the function call itself, but the queue most likely outlives that.

If you want to keep this function how it is, then it needs to be an unsafe function and the caller needs to guarantee that the buffer outlives the queue (and isn't used anywhere else as long as it is in the queue).